### ISSUE TITLE
Unexpected mongo exit code null. Restarting. Can't start Mongo server.  Meteor 2.6 & 2.6.1

### ISSUE THREAD

**Issue by paranoico** [meteor/meteor#11950]
Hello,

When creating a new project or upgrading to Meteor 2.6 or 2.6.1, an error arises:

Unexpected mongo exit code null. Restarting.  
Can't start Mongo server. 

So I had to downgrade again to 2.5.6

Could not find anything about this error in previous Meteor open bugs.

Thanks in advance.

**Comment by fredmaiaarantes** [comment]
Hey @paranoico, are you using the local embedded MongoDB?

If so, you can run `meteor reset` to remove your local database and cache. In Meteor 2.6 the embedded Mongo now uses MongoDB 5.0.5.

Be aware that this will remove your local data.

**Comment by paranoico** [comment]
Ola @fredmaiaarantes,

Yes, I am using the local embedded MongoDB.

And I have already tried `meteor reset` (found that solution in some very old Meteor bugs) but the problem persists. Even when making a new project.

I tried that on Meteor 2.6 and 2.6.1, same error results.

Obrigado.

**Comment by javoire** [comment]
I've been having this problem on WSL for about at least half a year (last I tried was around half a year ago)

```
$ meteor create myproject && cd myproject
$ meteor
[[[[[ ~/dev/myproject ]]]]]

=> Started proxy.
=> Started HMR server.
Unexpected mongo exit code null. Restarting.
Unexpected mongo exit code null. Restarting.
```
versions:
```
Ubuntu 20.04

uname -a
Linux DESKTOP-D4FJV90 4.4.0-19041-Microsoft #1237-Microsoft Sat Sep 11 14:32:00 PST 2021 x86_64 x86_64 x86_64 GNU/Linux

node -v
v16.13.0

meteor --version
Meteor 2.6.1

npm -v
8.1.0
```

**Comment by javoire** [comment]
_for WSL (linux on windows) users_

Update! This problem appears to have gone away for me for Ubuntu 20.04 by updating WSL from version 1 to 2.

Steps:

Check your WSL version(s) in PowerShell (run as Admin)
```
PS C:\WINDOWS\system32> wsl -l -v

  NAME            STATE           VERSION
* Ubuntu          Stopped         1
  Ubuntu-20.04    Stopped         1
```

Update/Install WSL 2 https://docs.microsoft.com/en-us/windows/wsl/install-manual

Set the version to WSL 2 for the Linux distro you're using:

```
PS C:\WINDOWS\system32> wsl --set-version Ubuntu-20.04 2
```

After the process finishes, version 2 should now be set for that distro:
```
PS C:\WINDOWS\system32> wsl -l -v

  NAME            STATE           VERSION
* Ubuntu          Stopped         1
  Ubuntu-20.04    Stopped         2
```

Creating and starting meteor works:

```
javoire@DESKTOP:~/dev$ meteor create myapp
Created a new Meteor app in 'myapp'.

...

javoire@DESKTOP:~/dev/myapp$ meteor
[[[[[ ~/dev/myapp]]]]]

=> Started proxy.
=> Started HMR server.
=> Started MongoDB.
=> Started your app.

=> App running at: http://localhost:3000/
```

**Comment by paranoico** [comment]
Hello,

@fredmaiaarantes Nop, not WIndows, I am using Ubuntu 20.04.4.

**Comment by YanFed** [comment]
Hello, I have the same problem. I'm using Linux Mint, based on Ubuntu 20. I tried to reset the project and other solutions I found on StockOverflow - nothing helps. I try to follow Meteor U course and can't start a new project, would it be with React or Blaze on the front end.

**Comment by paranoico** [comment]
Hi, have found some information regarding this issue and already done some testing:

Found these pages:
[https://forums.meteor.com/t/mongo-4-meteor-1-8/46065/7](https://forums.meteor.com/t/mongo-4-meteor-1-8/46065/7)
And
[https://github.com/meteor/meteor/issues/9591](https://github.com/meteor/meteor/issues/9591)

So I tried this command (from my `home` directory and after upgrading to Meteor 2.6.1):
`.meteor/packages/meteor-tool/2.6.1/mt-os.linux.x86_64/dev_bundle/mongodb/bin/mongod --dbpath /your/app/.meteor/local/db --repair`

But, I think have discovered the issue problem, the command does not work (!!!), it returns:
> Instrucción ilegal (`core' generado)

Same error arises even when trying:
`.meteor/packages/meteor-tool/2.6.1/mt-os.linux.x86_64/dev_bundle/mongodb/bin/mongod --help`

It seems the embedded mongodb server is corrupted or damaged.

One more thing, when creating a new app with Meteor 2.6.1, the `/your/app/.meteor/local/db` is empty, so it will never start.

Is there some way I can fix the embedded mongodb server???

Thanks.

**Comment by Avijobo** [comment]
Same problem here with new Meteor 2.7 project created.  I'm on Linux Mint 19.2 (Ubuntu 18.04)

**Comment by paranoico** [comment]
Hi @denihs 

I do not have a global installation of MongoDB, but we still have two apps in Meteor 1.5.x, we only use the embedded Mongo versions 3.x and 4.4.

The problem only arises on Meteor 2.6 and up to 2.7.1

How do we fix it in this case? How do we force a reinstall of embedded MongoDB on Meteor?

**Comment by denihs** [comment]
I believe it could be something related to the new Mongo version, and Meteor 2.6 introduced the integration with Mongo 5.

Could you run this command on your terminal `mongod --version`?

**Comment by paranoico** [comment]
`mongod --version` returns something like:

> mongod order not found, but you can install int with sudo apt install mongodb-server-core

Same message when executed on Meteor project dir.

**Comment by paranoico** [comment]
Hello,

After making some more research, i found the following articles explaining the exact same error message i have got ( `Illegal instruction     (core dumped)` ):

[MongoDB 5.0 requires a Sandy Bridge or newer CPU.](https://stackoverflow.com/questions/68392064/error-when-running-mongo-image-docker-entrypoint-sh-line-381)
[MongoDB Jira associated issue](https://jira.mongodb.org/browse/SERVER-54407)

So now it is clear that the error arises from within MongoDB embedded server.

We are going to stay at Meteor 2.5.6 until error is fixed.

I would like to recommend to Meteor Team avoid MongoDB 5 for now, or may be allow us to choose what version to use for the embedded server.

Thanks all for your help.

**Comment by Colear** [comment]
In the case of a VM (Proxmox for me) the solution provided in the first of the links above works perfectly : you just have to change the CPU type of the VM, by choosing for example host instead of the default value, and it works as soon as you reboot.

**Comment by sancaruso** [comment]
I have same problem here.
How do I can downgrade my application to Meteor 2.5.6 ?

**Comment by paranoico** [comment]
We created a new project targeting version 2.5.6:

`meteor create ProjectName --release 2.5.6`

Then copy files from previous  corrupted project dir and added the components again.

**Comment by paranoico** [comment]
Well, we use git for version control, so we just re-fetch the files again but may i thought you needed a more manual approach :smile:

**Comment by elregalo** [comment]
Hey guys, I just found about MeteorJs and thought giving it a try. I faced thesame issue when starting my app, becore I received "illigal instrution" when stating mongo InterR. I had to fork the process on Linux to start mongodb. One way to tackle this is to make sure your mongodb is runnng and start the app with ````MONGO_URL=mongodb://localhost:27017` meteor```

**Comment by paranoico** [comment]
Hello, we use the mongodb embedded engine inside Meteor so do not have it installed.

Moreover, our apps are distributed so all our customers have a local mongo instance installed. So we just cannot force everybody to buy new server just because mongo does not like old processors.

We also have some old development machines.

To me it is silly that mongo just requires Sandy Bridge now.

**Comment by tomswales** [comment]
Same issue for me using `Ubuntu 20.04.4 LTS`
Clean install of Meteor having completely removed all Meteor folders.
**Brand new project using meteor version 2.7.3**
`=> Started HMR server.                        
Unexpected mongo exit code null. Restarting.  
Unexpected mongo exit code null. Restarting.  
Unexpected mongo exit code null. Restarting.  
Can't start Mongo server.  `
**Old project using meteor version 2.5.1** 
`Started proxy.                             
=> Started HMR server.                        
=> Meteor 2.7.3 is available. Update this project with 'meteor update'.
=> Started MongoDB. `

**Comment by geekgonecrazy** [comment]
It swallowing the errors on this one is maddening to say the least.  

Spent several hours battling this today.  My 4.4 mongo container and using MONGO_URL worked fine.  But no matter what meteors wouldn’t.  Many other search results had me down path of locale.. but turns out it’s this and incompatible with my cpu.  🙈

> [MongoDB 5.0 requires a Sandy Bridge or newer CPU.](https://stackoverflow.com/questions/68392064/error-when-running-mongo-image-docker-entrypoint-sh-line-381) [MongoDB Jira associated issue](https://jira.mongodb.org/browse/SERVER-54407)
> 
> So now it is clear that the error arises from within MongoDB embedded server.

**Comment by tsiq-bradleyg** [comment]
I'm getting this issue with Meteor 2.8.1 on a Macbook Pro

**Comment by divs1210** [comment]
Hi, I'm running an existing project on `Ubuntu 24.04` on `VirtualBox` on a `Windows 11` host, and seeing this same issue.

This same project was running on Mac and on a Windows 10 machine under WSL.

Any pointers?

**Comment by Grubba27** [comment]
I do not think this is ongoing anymore. The root cause was MongoDB 5.0+'s AVX CPU requirement (upstream SERVER-54407), not a Meteor bug — on unsupported CPUs an external mongod via MONGO_URL works, and Meteor now prints mongod's stderr on startup failure. Please reopen if it persists.
